src/core/device.c

```diff
@@ -1075,6 +1079,103 @@ bool device_shall_be_bound_by(Unit *device, Unit *u) {
         return DEVICE(device)->bind_mounts;
 }
 
+/**
+ * While booting up it is likely that systemd-udevd is started after a couple of uevents were already
+ * triggered by the kernel and are missed for that reason. To fix that up device_start() is used to
+ * re-trigger the corresponding uevent of the particular device unit.
+ *
+ * The re-trigger is only done in case the device is not following another device, was never in plugged
+ * state before, is in the dead state, but is actually plugged (corresponding sysfs entries are existent).
+ *
+ * Note: This accounts only for device units which are explicitly referenced as dependency by other units
+ *       and are tagged in udev rules (TAG+="systemd").
+ *
+ *       This might only be useful in highly embedded environments, e.g. automotive control units, where
+ *       existent devices and the whole dependency tree are known, the ecosystem is well-defined and startup
+ *       performance matters.
+ *
+ *       For desktop and server environments systemd-udev-trigger.service should still be used.
+ */
+static int device_start(Unit *u) {
+        Device *d = DEVICE(u);
+        _cleanup_(sd_device_unrefp) sd_device *dev = NULL;
+        _cleanup_free_ char *path = NULL;
+        int ret;
+
+        if (device_following(u))
+                goto done;
+
+        if (!(d->state == DEVICE_DEAD))
+                goto done;
+
+        if (d->plugged_once)
+                goto done;
+
+        ret = unit_name_to_path(u->id, &path);
+        if (ret) {
+                log_unit_warning(u, "Failed to get device path from unit: \"%s\"", u->id);
```

Comment:
`return log_unit_warning_errno(u, SYNTHETIC_ERRNO(EBADR), "Failed...` here and in all other error paths

Reply:
This would also get rid of the extra log in the error case - intended?

Reply:
Can't the two messages be unified?

Reply:
> Can't the two messages be unified?

In a less verbose way, sure:
```suggestion
                return log_unit_warning_errno(u,SYNTETHIC_ERRNO(EBADR),
                                "Failed to get device path from unit: \"%s\" (%d)", u->id, r);
```